examples/ecs/relationships.rs

```diff
@@ -53,12 +53,14 @@ fn main() {
         // Relations are just components, so we can add them into the bundle that we're spawning.
         let bob = commands.spawn((Name::new("Bob"), Targeting(alice))).id();
 
-        // The `with_related` helper method on `EntityCommands` can be used to add relations in a more ergonomic way.
+        // The `with_related` and `with_relationships` helper methods on `EntityCommands` can be used to add relations in a more ergonomic way.
         let charlie = commands
             .spawn((Name::new("Charlie"), Targeting(bob)))
-            // The `with_related` method will automatically add the `Targeting` component to any entities spawned within the closure,
+            // The `with_related` method will spawn a bundle with `Targeting` relationship
+            .with_related::<Targeting>(Name::new("James"))
+            // The `with_relationships` method will automatically add the `Targeting` component to any entities spawned within the closure,
             // targeting the entity that we're calling `with_related` on.
-            .with_related::<Targeting>(|related_spawner_commands| {
+            .with_relationships::<Targeting>(|related_spawner_commands| {
```

Comment:
This is a good example of why this (and the original) naming feels so unclear to me. `with_children` clearly implied that "this is used to generate new entities that are related to the parent". Neither `with_related` or `with_relationships` communicates that to me.

`with_related_entities` would be much clearer at a glace, if more verbose. Something like `.spawn_and_relate` would be very clearer, if a bit weird, since it's not actually doing the spawning. `relate_spawned_entities` might make more sense.

Overall `with_related_entities` is winning out for me. I think that we should give the `with_related` name to the single entity form though!

Reply:
The method we're discussing here doesn't take a bundle though: it takes a closure that can be used to spawn additional entities and automatically link them.

Reply:
I agree that `with_related` should got to the single entity case. `with_related_entities` doesn't feel great, but I also can't come up with anything I like better. Given that in general `with_X` is likely to lose prominence relative to direct spawning APIs (`bsn!`, `children!`, `related!`, etc), I'm cool with the name quality tradeoff here.